### PR TITLE
Fix the failures on VK.pipeline.monolithic.multisample.mixed_attachme…

### PR DESCRIPTION
**Pull request by ruiminzhao**
…nt_samples.shader_builtins.* cases

The failures are caused by the regression from commit-id: d6d61e6b5baba To fix this issue, it can only broadcast sample mask when the depthstencil test is disable.

### CODE REVIEW

**Comment by amdvlk-admin** [comment]
### Test summary for commit d633fcadfd87f88392d1019b76281be7f4699f90
<details><summary>CTS tests (Failed: 0/209067)</summary><ul><li>Built with version 1.3.3.1</li>
<details open><summary>Rhel 8.9, Gfx10</summary><ul>

<li>Passed: 34736/69788 (49.8%)</li>
<li>Failed: 0/69788 (0.0%)</li>
<li>Not Supported: 35052/69788 (50.2%)</li><li>Warnings: 0/69788 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 38620/69504 (55.6%)</li>
<li>Failed: 0/69504 (0.0%)</li>
<li>Not Supported: 30884/69504 (44.4%)</li><li>Warnings: 0/69504 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Navi2x</summary><ul>

<li>Passed: 38654/69775 (55.4%)</li>
<li>Failed: 0/69775 (0.0%)</li>
<li>Not Supported: 31121/69775 (44.6%)</li><li>Warnings: 0/69775 (0.0%)</li>
</ul></details>


</ul></details>

File: lgc/state/PipelineState.cpp [hunk 1]
```diff
@@ -1295,6 +1299,22 @@ void PipelineState::readGraphicsState(Module *module) {
     m_rasterizerState.perSampleShading |= 1;
 }
 
+// =====================================================================================================================
+// Record depthsenticil state into the IR metadata
```
Comment (nhaehnle) [inline lgc/state/PipelineState.cpp:1303]: typo: depthstencil

File: lgc/state/PipelineState.cpp [hunk 2]
```diff
@@ -1295,6 +1299,22 @@ void PipelineState::readGraphicsState(Module *module) {
...
+//
+// @param [in/out] module : IR module to record into
+void PipelineState::recordDepthStencilState(Module *module) {
+  setNamedMetadataToArrayOfInt32(module, m_depthStencilState, DepthStencilStateMetadataName);
+}
+
+// =====================================================================================================================
+// Read depthsenticil state from the IR metadata
```
Comment (nhaehnle) [inline lgc/state/PipelineState.cpp:1311]: also here

**Review by nhaehnle — COMMENTED** [review]
This is a bit surprising. The new `else` path is equivalent to forcing `pixelShaderSamples == numSamples`. Does depth-test enable really force the PS to run at sample rate in hardware? What is the logic at the GRBM register level?

**Comment by amdvlk-admin** [comment]
### Test summary for commit 4fd0260587fac82a78822b331a4c3cc521db5c7f
<details><summary>CTS tests (Failed: 0/209351)</summary><ul><li>Built with version 1.3.3.1</li>
<details open><summary>Rhel 9.0, Gfx10</summary><ul>

<li>Passed: 34735/69788 (49.8%)</li>
<li>Failed: 0/69788 (0.0%)</li>
<li>Not Supported: 35053/69788 (50.2%)</li><li>Warnings: 0/69788 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 38620/69788 (55.3%)</li>
<li>Failed: 0/69788 (0.0%)</li>
<li>Not Supported: 31168/69788 (44.7%)</li><li>Warnings: 0/69788 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Navi2x</summary><ul>

<li>Passed: 38654/69775 (55.4%)</li>
<li>Failed: 0/69775 (0.0%)</li>
<li>Not Supported: 31121/69775 (44.6%)</li><li>Warnings: 0/69775 (0.0%)</li>
</ul></details>


</ul></details>

**Comment by amdvlk-admin** [comment]
### Test summary for commit f6a5210b00c5157495ed38e81ff82dab87ab4b92
<details><summary>CTS tests (Failed: 0/209351)</summary><ul><li>Built with version 1.3.3.1</li>
<details open><summary>Rhel 9.0, Gfx10</summary><ul>

<li>Passed: 34735/69788 (49.8%)</li>
<li>Failed: 0/69788 (0.0%)</li>
<li>Not Supported: 35053/69788 (50.2%)</li><li>Warnings: 0/69788 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 38620/69788 (55.3%)</li>
<li>Failed: 0/69788 (0.0%)</li>
<li>Not Supported: 31168/69788 (44.7%)</li><li>Warnings: 0/69788 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Navi2x</summary><ul>

<li>Passed: 38654/69775 (55.4%)</li>
<li>Failed: 0/69775 (0.0%)</li>
<li>Not Supported: 31121/69775 (44.6%)</li><li>Warnings: 0/69775 (0.0%)</li>
</ul></details>


</ul></details>

**Comment by amdvlk-admin** [comment]
### Test summary for commit af525d2e9ad2a86a1cb2d8b17f7ed35cc03176e0
<details><summary>CTS tests (Failed: 0/209067)</summary><ul><li>Built with version 1.3.3.1</li>
<details open><summary>Rhel 9.0, Gfx10</summary><ul>

<li>Passed: 34735/69788 (49.8%)</li>
<li>Failed: 0/69788 (0.0%)</li>
<li>Not Supported: 35053/69788 (50.2%)</li><li>Warnings: 0/69788 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 38620/69504 (55.6%)</li>
<li>Failed: 0/69504 (0.0%)</li>
<li>Not Supported: 30884/69504 (44.4%)</li><li>Warnings: 0/69504 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Navi2x</summary><ul>

<li>Passed: 38654/69775 (55.4%)</li>
<li>Failed: 0/69775 (0.0%)</li>
<li>Not Supported: 31121/69775 (44.6%)</li><li>Warnings: 0/69775 (0.0%)</li>
</ul></details>


</ul></details>

**Comment by amdvlk-admin** [comment]
### Test summary for commit 7f51bad66780a336fc24a67879133e15d99785d2
<details><summary>CTS tests (Failed: 0/209351)</summary><ul><li>Built with version 1.3.3.1</li>
<details open><summary>Rhel 8.9, Gfx10</summary><ul>

<li>Passed: 34736/69788 (49.8%)</li>
<li>Failed: 0/69788 (0.0%)</li>
<li>Not Supported: 35052/69788 (50.2%)</li><li>Warnings: 0/69788 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 38620/69788 (55.3%)</li>
<li>Failed: 0/69788 (0.0%)</li>
<li>Not Supported: 31168/69788 (44.7%)</li><li>Warnings: 0/69788 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Navi2x</summary><ul>

<li>Passed: 38654/69775 (55.4%)</li>
<li>Failed: 0/69775 (0.0%)</li>
<li>Not Supported: 31121/69775 (44.6%)</li><li>Warnings: 0/69775 (0.0%)</li>
</ul></details>


</ul></details>

**Comment by amdvlk-admin** [comment]
### Test summary for commit 20ce77e8dded94c892c3ac267bf022690811e08d
<details><summary>CTS tests (Failed: 0/209351)</summary><ul><li>Built with version 1.3.3.1</li>
<details open><summary>Rhel 9.0, Gfx10</summary><ul>

<li>Passed: 34735/69788 (49.8%)</li>
<li>Failed: 0/69788 (0.0%)</li>
<li>Not Supported: 35053/69788 (50.2%)</li><li>Warnings: 0/69788 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 38620/69788 (55.3%)</li>
<li>Failed: 0/69788 (0.0%)</li>
<li>Not Supported: 31168/69788 (44.7%)</li><li>Warnings: 0/69788 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Navi2x</summary><ul>

<li>Passed: 38654/69775 (55.4%)</li>
<li>Failed: 0/69775 (0.0%)</li>
<li>Not Supported: 31121/69775 (44.6%)</li><li>Warnings: 0/69775 (0.0%)</li>
</ul></details>


</ul></details>

**Comment by nhaehnle** [comment]
> This is a bit surprising. The new `else` path is equivalent to forcing `pixelShaderSamples == numSamples`. Does depth-test enable really force the PS to run at sample rate in hardware? What is the logic at the GRBM register level?

What about this?

**Comment by ruiminzhao** [comment]
> > This is a bit surprising. The new `else` path is equivalent to forcing `pixelShaderSamples == numSamples`. Does depth-test enable really force the PS to run at sample rate in hardware? What is the logic at the GRBM register level?
> 
> What about this?

This fix is ported from SCPC, which has been fixed on SWDEV-320599. I will discuss with @qiaojbao about the details when back to office tomorrow.

**Comment by ruiminzhao** [comment]
> > > This is a bit surprising. The new `else` path is equivalent to forcing `pixelShaderSamples == numSamples`. Does depth-test enable really force the PS to run at sample rate in hardware? What is the logic at the GRBM register level?
> > 
> > 
> > What about this?
> 
> This fix is ported from SCPC, which has been fixed on SWDEV-320599. I will discuss with @qiaojbao about the details when back to office tomorrow.

After discussion, it's just a workaround on SCPC:
"this change will cause vk cts failure (dEQP-VK.pipeline.multisample.mixed_attachment_samples.shader_builtins.coverage_4_color_2_depth_stencil_4.r8g8b8a8_unorm_d16_unorm), the difference between vkcts and glescts above is glcts's depthTestEnable is false, vkcts is true.
So add depthTestEnable judgment first."

I have dump related log and it needs more investigation to find the root cause on this. I will raise another PR to investigate on this not only LLPC but also on SCPC.
This is only workaround on LLPC keeping same with SCPC which will only fix the regression and block on CTS tests

**Comment by amdvlk-admin** [comment]
### Test summary for commit b72bcc51c0b11780cae520842b53ac6c7bcd8c82
<details><summary>CTS tests (Failed: 0/209351)</summary><ul><li>Built with version 1.3.5.2</li>
<details open><summary>Rhel 8.9, Gfx10</summary><ul>

<li>Passed: 36979/69788 (53.0%)</li>
<li>Failed: 0/69788 (0.0%)</li>
<li>Not Supported: 32809/69788 (47.0%)</li><li>Warnings: 0/69788 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 41144/69788 (59.0%)</li>
<li>Failed: 0/69788 (0.0%)</li>
<li>Not Supported: 28644/69788 (41.0%)</li><li>Warnings: 0/69788 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Navi2x</summary><ul>

<li>Passed: 41178/69775 (59.0%)</li>
<li>Failed: 0/69775 (0.0%)</li>
<li>Not Supported: 28597/69775 (41.0%)</li><li>Warnings: 0/69775 (0.0%)</li>
</ul></details>


</ul></details>

**Comment by amdvlk-admin** [comment]
### Test summary for commit 22d46a2942a262495a89aee5b379581b1f607cb7
<details><summary>CTS tests (Failed: 0/209067)</summary><ul><li>Built with version 1.3.5.2</li>
<details open><summary>Rhel 9.0, Gfx10</summary><ul>

<li>Passed: 36978/69788 (53.0%)</li>
<li>Failed: 0/69788 (0.0%)</li>
<li>Not Supported: 32810/69788 (47.0%)</li><li>Warnings: 0/69788 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 41144/69504 (59.2%)</li>
<li>Failed: 0/69504 (0.0%)</li>
<li>Not Supported: 28360/69504 (40.8%)</li><li>Warnings: 0/69504 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Navi2x</summary><ul>

<li>Passed: 41178/69775 (59.0%)</li>
<li>Failed: 0/69775 (0.0%)</li>
<li>Not Supported: 28597/69775 (41.0%)</li><li>Warnings: 0/69775 (0.0%)</li>
</ul></details>


</ul></details>

**Comment by amdvlk-admin** [comment]
### Test summary for commit 49e3fee31bc3df13aedc69f02734bca56c0183c0
<details><summary>CTS tests (Failed: 0/209067)</summary><ul><li>Built with version 1.3.5.2</li>
<details open><summary>Rhel 9.0, Gfx10</summary><ul>

<li>Passed: 36978/69788 (53.0%)</li>
<li>Failed: 0/69788 (0.0%)</li>
<li>Not Supported: 32810/69788 (47.0%)</li><li>Warnings: 0/69788 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 41144/69504 (59.2%)</li>
<li>Failed: 0/69504 (0.0%)</li>
<li>Not Supported: 28360/69504 (40.8%)</li><li>Warnings: 0/69504 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Navi2x</summary><ul>

<li>Passed: 41178/69775 (59.0%)</li>
<li>Failed: 0/69775 (0.0%)</li>
<li>Not Supported: 28597/69775 (41.0%)</li><li>Warnings: 0/69775 (0.0%)</li>
</ul></details>


</ul></details>

**Comment by amdvlk-admin** [comment]
### Test summary for commit 002c14e6940464be21e0a55c6737b02a465490fc
<details><summary>CTS tests (Failed: 0/209067)</summary><ul><li>Built with version 1.3.5.2</li>
<details open><summary>Rhel 9.0, Gfx10</summary><ul>

<li>Passed: 36978/69788 (53.0%)</li>
<li>Failed: 0/69788 (0.0%)</li>
<li>Not Supported: 32810/69788 (47.0%)</li><li>Warnings: 0/69788 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 41144/69504 (59.2%)</li>
<li>Failed: 0/69504 (0.0%)</li>
<li>Not Supported: 28360/69504 (40.8%)</li><li>Warnings: 0/69504 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Navi2x</summary><ul>

<li>Passed: 41178/69775 (59.0%)</li>
<li>Failed: 0/69775 (0.0%)</li>
<li>Not Supported: 28597/69775 (41.0%)</li><li>Warnings: 0/69775 (0.0%)</li>
</ul></details>


</ul></details>

**Comment by nhaehnle** [comment]
> After discussion, it's just a workaround on SCPC: "this change will cause vk cts failure (dEQP-VK.pipeline.multisample.mixed_attachment_samples.shader_builtins.coverage_4_color_2_depth_stencil_4.r8g8b8a8_unorm_d16_unorm), the difference between vkcts and glescts above is glcts's depthTestEnable is false, vkcts is true. So add depthTestEnable judgment first."
> 
> I have dump related CTS case:
> 
> ```
> layout(location = 0) out vec4 o_color;
> 
> void main(void)
> {
>     vec4 col = vec4(0.0, 0.0, 0.0, 1.0);
> 
>     if (gl_SampleMaskIn[0] == (1 << gl_SampleID))
>         col.g = 1.0;
>     else
>         col.r = 1.0;
> 
>     if (gl_SampleID >= 2)  // number of color samples, should not happen
>         col.b = 1.0;
> 
>     o_color = col;
> }
> ```
> 
> It seems gl_SampleMaskIn[0] is calculated differently between SampleMask broadcasting and non-SampleMask broadcasting. So it maybe impact on the col.g and col.r value.

Thanks. It looks like probably the real root cause is that this pixel shader uses gl_SampleID and should therefore run at sample rate (`builtInUsage.runAtSampleRate`) which takes precedence over `pixelShaderSamples.

So really, I suspect the correct fix is:
```c++
       unsigned baseMask = 1;
+      if (!builtInUsage.runAtSampleRate) {
!        unsigned baseMaskSamples = m_pipelineState->getRasterizerState().pixelShaderSamples;
!        while (baseMaskSamples < m_pipelineState->getRasterizerState().numSamples) {
!          baseMask |= baseMask << baseMaskSamples;
!          baseMaskSamples *= 2;
!        }
+      }
```

**Comment by ruiminzhao** [comment]
> > After discussion, it's just a workaround on SCPC: "this change will cause vk cts failure (dEQP-VK.pipeline.multisample.mixed_attachment_samples.shader_builtins.coverage_4_color_2_depth_stencil_4.r8g8b8a8_unorm_d16_unorm), the difference between vkcts and glescts above is glcts's depthTestEnable is false, vkcts is true. So add depthTestEnable judgment first."
> > I have dump related CTS case:
> > ```
> > layout(location = 0) out vec4 o_color;
> > 
> > void main(void)
> > {
> >     vec4 col = vec4(0.0, 0.0, 0.0, 1.0);
> > 
> >     if (gl_SampleMaskIn[0] == (1 << gl_SampleID))
> >         col.g = 1.0;
> >     else
> >         col.r = 1.0;
> > 
> >     if (gl_SampleID >= 2)  // number of color samples, should not happen
> >         col.b = 1.0;
> > 
> >     o_color = col;
> > }
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > It seems gl_SampleMaskIn[0] is calculated differently between SampleMask broadcasting and non-SampleMask broadcasting. So it maybe impact on the col.g and col.r value.
> 
> Thanks. It looks like probably the real root cause is that this pixel shader uses gl_SampleID and should therefore run at sample rate (`builtInUsage.runAtSampleRate`) which takes precedence over `pixelShaderSamples.
> 
> So really, I suspect the correct fix is:
> 
> ```c++
>        unsigned baseMask = 1;
> +      if (!builtInUsage.runAtSampleRate) {
> !        unsigned baseMaskSamples = m_pipelineState->getRasterizerState().pixelShaderSamples;
> !        while (baseMaskSamples < m_pipelineState->getRasterizerState().numSamples) {
> !          baseMask |= baseMask << baseMaskSamples;
> !          baseMaskSamples *= 2;
> !        }
> +      }
> ```

Thanks Nicolai, I think this fix is closer to final solution. 

But after some verification with this fix, the regression tests(dEQP-VK.pipeline.monolithic.multisample.mixed_attachment_samples.shader_builtins.coverage_16_color_8_depth_stencil) can be passed as ```runAtSampleRate == 1 && perSampleShading = 1" and the mask is 1<<gl_sampleId

However, the original cases(dEQP-VK.pipeline.fast_linked_library.multisample_shader_builtin.sample_mask.bit_count_0_5.137_191_1.samples_4) have been failed as ```runAtSampleRate == 1 && perSampleShading = 1 ```...... So I find this variable is set in PatchResourceCollect::processShader :
```
  if (m_shaderStage == ShaderStageFragment) {
    if (m_pipelineState->getRasterizerState().perSampleShading) {
      if (m_resUsage->builtInUsage.fs.fragCoord || m_resUsage->builtInUsage.fs.pointCoord ||
          m_resUsage->builtInUsage.fs.sampleMaskIn || m_resUsage->resourceWrite)
        m_resUsage->builtInUsage.fs.runAtSampleRate = true;
    }
```
 Maybe this part code also needs update?

**Comment by amdvlk-admin** [comment]
### Test summary for commit 621e3c72b48a596ba6cf62d308f71e5a0a29ddf4
<details><summary>CTS tests (Failed: 24/209351)</summary><ul><li>Built with version 1.3.5.2</li>
<details open><summary>Rhel 9.0, Gfx10</summary><ul>

<li>Passed: 36978/69788 (53.0%)</li>
<li>Failed: 0/69788 (0.0%)</li>
<li>Not Supported: 32810/69788 (47.0%)</li><li>Warnings: 0/69788 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 41130/69788 (58.9%)</li>
<li><details><summary>Failed: 14/69788 (0.0%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.fragment_shading_rate.dynamic_rendering.primary_cmd_buff.monolithic.samplemaskin.dynamic.attachment.noshaderrate.keep.replace.33x35.samples1.vs
Stack trace: Timeout

FAILURE: dEQP-VK.fragment_shading_rate.dynamic_rendering.primary_cmd_buff.monolithic.samplemaskin.dynamic.attachment.noshaderrate.replace.keep.151x431.samples1.ms
Stack trace: Timeout

FAILURE: dEQP-VK.fragment_shading_rate.dynamic_rendering.primary_cmd_buff.monolithic.samplemaskin.dynamic.attachment.shaderrate.replace.keep.1x1.samples1.ms
Stack trace: Timeout

FAILURE: dEQP-VK.fragment_shading_rate.dynamic_rendering.primary_cmd_buff.monolithic.samplemaskin.dynamic.attachment_noimageview.noshaderrate.keep.replace.151x431.samples2.vs
Stack trace: Fail

FAILURE: dEQP-VK.fragment_shading_rate.dynamic_rendering.primary_cmd_buff.monolithic.samplemaskin.dynamic.noattachment.noshaderrate.keep.keep.4x4.samples2.ms
Stack trace: Fail

FAILURE: dEQP-VK.fragment_shading_rate.dynamic_rendering.primary_cmd_buff.monolithic.samplemaskin.dynamic.noattachment.noshaderrate.replace.keep.33x35.samples1.ms
Stack trace: Timeout

FAILURE: dEQP-VK.fragment_shading_rate.dynamic_rendering.primary_cmd_buff.monolithic.samplemaskin.dynamic.noattachment.shaderrate.keep.keep.256x256.samples4.ms
Stack trace: Fail

FAILURE: dEQP-VK.fragment_shading_rate.renderpass2.monolithic.samplemaskin.dynamic.attachment.noshaderrate.replace.replace.151x431.samples1.vs
Stack trace: Timeout

FAILURE: dEQP-VK.fragment_shading_rate.renderpass2.monolithic.samplemaskin.dynamic.noattachment.shaderrate.replace.keep.33x35.samples4.ms
Stack trace: Fail

FAILURE: dEQP-VK.fragment_shading_rate.renderpass2.monolithic.samplemaskin.dynamic.noattachment.shaderrate.replace.replace.256x256.samples2.vs
Stack trace: Fail

FAILURE: dEQP-VK.fragment_shading_rate.renderpass2.monolithic.samplemaskin.dynamic.noattachmentptr.noshaderrate.keep.keep.33x35.samples1.vs
Stack trace: Timeout

FAILURE: dEQP-VK.fragment_shading_rate.renderpass2.monolithic.samplemaskin.dynamic.noattachmentptr.shaderrate.keep.keep.1x1.samples2.vs
Stack trace: Fail

FAILURE: dEQP-VK.fragment_shading_rate.renderpass2.monolithic.samplemaskin.dynamic.noattachmentptr.shaderrate.replace.replace.1x1.samples1.ms
Stack trace: Timeout

FAILURE: dEQP-VK.fragment_shading_rate.renderpass2.monolithic.samplemaskin.dynamic.noattachmentptr.shaderrate.replace.replace.1x1.samples1.vs
Stack trace: Timeout
```
</p></details></li>

<li>Not Supported: 28644/69788 (41.0%)</li><li>Warnings: 0/69788 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Navi2x</summary><ul>

<li>Passed: 41168/69775 (59.0%)</li>
<li><details><summary>Failed: 10/69775 (0.0%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.fragment_shading_rate.dynamic_rendering.primary_cmd_buff.monolithic.samplemaskin.dynamic.attachment.noshaderrate.keep.replace.33x35.samples1.vs
Stack trace: Timeout

FAILURE: dEQP-VK.fragment_shading_rate.dynamic_rendering.primary_cmd_buff.monolithic.samplemaskin.dynamic.attachment.noshaderrate.replace.keep.151x431.samples1.ms
Stack trace: Timeout

FAILURE: dEQP-VK.fragment_shading_rate.dynamic_rendering.primary_cmd_buff.monolithic.samplemaskin.dynamic.attachment.shaderrate.replace.keep.1x1.samples1.ms
Stack trace: Timeout

FAILURE: dEQP-VK.fragment_shading_rate.dynamic_rendering.primary_cmd_buff.monolithic.samplemaskin.dynamic.attachment_noimageview.noshaderrate.keep.replace.151x431.samples2.vs
Stack trace: Fail

FAILURE: dEQP-VK.fragment_shading_rate.dynamic_rendering.primary_cmd_buff.monolithic.samplemaskin.dynamic.noattachment.noshaderrate.replace.keep.33x35.samples1.ms
Stack trace: Timeout

FAILURE: dEQP-VK.fragment_shading_rate.renderpass2.monolithic.samplemaskin.dynamic.attachment.noshaderrate.replace.replace.151x431.samples1.vs
Stack trace: Timeout

FAILURE: dEQP-VK.fragment_shading_rate.renderpass2.monolithic.samplemaskin.dynamic.noattachment.shaderrate.replace.replace.256x256.samples2.vs
Stack trace: Fail

FAILURE: dEQP-VK.fragment_shading_rate.renderpass2.monolithic.samplemaskin.dynamic.noattachmentptr.noshaderrate.keep.keep.33x35.samples1.vs
Stack trace: Timeout

FAILURE: dEQP-VK.fragment_shading_rate.renderpass2.monolithic.samplemaskin.dynamic.noattachmentptr.shaderrate.replace.replace.1x1.samples1.ms
Stack trace: Timeout

FAILURE: dEQP-VK.fragment_shading_rate.renderpass2.monolithic.samplemaskin.dynamic.noattachmentptr.shaderrate.replace.replace.1x1.samples1.vs
Stack trace: Timeout
```
</p></details></li>

<li>Not Supported: 28597/69775 (41.0%)</li><li>Warnings: 0/69775 (0.0%)</li>
</ul></details>


</ul></details>

**Comment by amdvlk-admin** [comment]
### Test summary for commit ae0b01bb27936c46b63cae1159689aca1d425b0f
<details><summary>CTS tests (Failed: 0/209067)</summary><ul><li>Built with version 1.3.5.2</li>
<details open><summary>Rhel 9.0, Gfx10</summary><ul>

<li>Passed: 36978/69788 (53.0%)</li>
<li>Failed: 0/69788 (0.0%)</li>
<li>Not Supported: 32810/69788 (47.0%)</li><li>Warnings: 0/69788 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 41144/69504 (59.2%)</li>
<li>Failed: 0/69504 (0.0%)</li>
<li>Not Supported: 28360/69504 (40.8%)</li><li>Warnings: 0/69504 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Navi2x</summary><ul>

<li>Passed: 41178/69775 (59.0%)</li>
<li>Failed: 0/69775 (0.0%)</li>
<li>Not Supported: 28597/69775 (41.0%)</li><li>Warnings: 0/69775 (0.0%)</li>
</ul></details>


</ul></details>

**Comment by amdvlk-admin** [comment]
### Test summary for commit fe3917aeeaaaf2f3c2f9d7e89e524ac17e9c1830
<details><summary>CTS tests (Failed: 0/209351)</summary><ul><li>Built with version 1.3.5.2</li>
<details open><summary>Rhel 9.0, Gfx10</summary><ul>

<li>Passed: 36978/69788 (53.0%)</li>
<li>Failed: 0/69788 (0.0%)</li>
<li>Not Supported: 32810/69788 (47.0%)</li><li>Warnings: 0/69788 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 41144/69788 (59.0%)</li>
<li>Failed: 0/69788 (0.0%)</li>
<li>Not Supported: 28644/69788 (41.0%)</li><li>Warnings: 0/69788 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Navi2x</summary><ul>

<li>Passed: 41178/69775 (59.0%)</li>
<li>Failed: 0/69775 (0.0%)</li>
<li>Not Supported: 28597/69775 (41.0%)</li><li>Warnings: 0/69775 (0.0%)</li>
</ul></details>


</ul></details>

**Comment by piotrAMD** [comment]
Thanks, that looks reasonable. The final version with "!builtInUsage.sampleId" is in the spirit of Nicolai's proposal.

**Review by dstutt — APPROVED** [review]
LGTM

**Comment by amdvlk-admin** [comment]
### Test summary for commit 8f7fcd5cc27b87cf39a45831759591d265313946
<details><summary>CTS tests (Failed: 0/208496)</summary><ul><li>Built with version 1.3.5.2</li>
<details open><summary>Rhel 9.0, Gfx10</summary><ul>

<li>Passed: 36978/69503 (53.2%)</li>
<li>Failed: 0/69503 (0.0%)</li>
<li>Not Supported: 32525/69503 (46.8%)</li><li>Warnings: 0/69503 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 41144/69503 (59.2%)</li>
<li>Failed: 0/69503 (0.0%)</li>
<li>Not Supported: 28359/69503 (40.8%)</li><li>Warnings: 0/69503 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Navi2x</summary><ul>

<li>Passed: 41178/69490 (59.3%)</li>
<li>Failed: 0/69490 (0.0%)</li>
<li>Not Supported: 28312/69490 (40.7%)</li><li>Warnings: 0/69490 (0.0%)</li>
</ul></details>


</ul></details>

**Comment by amdvlk-admin** [comment]
### Test summary for commit 09d49a4d2545771abc70a0b2526c8213b8c901f7
<details><summary>CTS tests (Failed: 0/208496)</summary><ul><li>Built with version 1.3.5.2</li>
<details open><summary>Rhel 9.0, Gfx10</summary><ul>

<li>Passed: 36978/69503 (53.2%)</li>
<li>Failed: 0/69503 (0.0%)</li>
<li>Not Supported: 32525/69503 (46.8%)</li><li>Warnings: 0/69503 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 41144/69503 (59.2%)</li>
<li>Failed: 0/69503 (0.0%)</li>
<li>Not Supported: 28359/69503 (40.8%)</li><li>Warnings: 0/69503 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Navi2x</summary><ul>

<li>Passed: 41178/69490 (59.3%)</li>
<li>Failed: 0/69490 (0.0%)</li>
<li>Not Supported: 28312/69490 (40.7%)</li><li>Warnings: 0/69490 (0.0%)</li>
</ul></details>


</ul></details>

**Comment by amdvlk-admin** [comment]
### Test summary for commit 19ebcc091936defc5ae7830fb503675008023c20
<details><summary>CTS tests (Failed: 0/208496)</summary><ul><li>Built with version 1.3.5.2</li>
<details open><summary>Rhel 9.0, Gfx10</summary><ul>

<li>Passed: 36978/69503 (53.2%)</li>
<li>Failed: 0/69503 (0.0%)</li>
<li>Not Supported: 32525/69503 (46.8%)</li><li>Warnings: 0/69503 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 41144/69503 (59.2%)</li>
<li>Failed: 0/69503 (0.0%)</li>
<li>Not Supported: 28359/69503 (40.8%)</li><li>Warnings: 0/69503 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Navi2x</summary><ul>

<li>Passed: 41178/69490 (59.3%)</li>
<li>Failed: 0/69490 (0.0%)</li>
<li>Not Supported: 28312/69490 (40.7%)</li><li>Warnings: 0/69490 (0.0%)</li>
</ul></details>


</ul></details>

**Pull request merged**